Source/JavaScriptCore/heap/PreciseAllocation.cpp

```diff
@@ -121,9 +120,30 @@ PreciseAllocation* PreciseAllocation::tryReallocate(size_t size, Subspace* subsp
     unsigned oldAdjustment = m_adjustment;
     void* oldBasePointer = basePointer();
 
-    void* newSpace = subspace->alignedMemoryAllocator()->tryReallocateMemory(oldBasePointer, adjustedAlignmentAllocationSize);
-    if (!newSpace)
-        return nullptr;
+    void* newSpace = nullptr;
+    size_t pageSize = WTF::pageSize();
+    info.usePageZeroing = newAllocationSize > 5 * pageSize;
```

Comment:
I would increase this much more since pageSize rounding can cause wastage. Like, 16MB or more.